File: clients/csharp/Model/ExplosionParameters.cs
namespace AiCup2019.Model
{
    public struct ExplosionParameters
    {
        public double Radius { get; set; }
        public int Damage { get; set; }
        public ExplosionParameters(double radius, int damage)
        {
            this.Radius = radius;
            this.Damage = damage;
        }
        public static ExplosionParameters ReadFrom(System.IO.BinaryReader reader)
        {
            var result = new ExplosionParameters();
            result.Radius = reader.ReadDouble();
            result.Damage = reader.ReadInt32();
            return result;
        }
        public void WriteTo(System.IO.BinaryWriter writer)
        {
            writer.Write(Radius);
            writer.Write(Damage);
        }
    }
}

File: clients/fsharp/Model/ExplosionParameters.fs
namespace AiCup2019.Model

type ExplosionParameters = {Radius : double; Damage: int} with
    member this.writeTo (writer: System.IO.BinaryWriter) =
        writer.Write this.Radius
        writer.Write this.Damage

    static member readFrom (reader: System.IO.BinaryReader) = 
        {Radius = reader.ReadDouble(); Damage = reader.ReadInt32()}
